plugins/BEdita/API/src/Controller/LoginController.php

```diff
@@ -167,21 +169,59 @@ protected function jwtTokens(array $user)
      * Read logged user data.
      *
      * @return void
-     * @throws \Cake\Network\Exception\UnauthorizedException Throws an exception if user not logged.
      */
     public function whoami()
     {
         $this->request->allowMethod('get');
 
+        $user = $this->userEntity();
+
+        $this->set(compact('user'));
+        $this->set('_serialize', ['user']);
+    }
+
+    /**
+     * Update user profile data.
+     *
+     * @return void
+     * @throws \Cake\Network\Exception\BadRequestException On invalid input data
+     */
+    public function update()
+    {
+        $this->request->allowMethod('patch');
+
+        $notAllowed = ['username', 'password', 'password_hash', 'email'];
+        $data = $this->request->getData();
+        if (!empty(array_intersect($notAllowed, array_keys($data)))) {
+            throw new BadRequestException([
+                'title' => __d('bedita', 'Bad input data'),
+                'detail' => sprintf('Fields not allowed: %s', implode(', ', $notAllowed)),
+            ]);
+        }
```

Comment:
Instead of this approach, I'd prefer to use entity's accessible fields. So, `$entity->setAccessible('username', false);` and so on. However, no bad request exception would be thrown in case a user tries to update one of those fields, and their values would be silently discarded. This is consistent with the behaviour in other parts of the application (i.e.: `PATCH /objects/123` with a non-accessible field), but if we want an error to be raised in this case we could move this logic to the `SaveEntityAction`.

We might evaluate this refactor in the future, though.

Reply:
That's right but this behavior is `enpoint` dependent: from `/users` we should be able to modify those properties, but from `/auth/user` not. 

As whether or not throw an exception: generally I'd prefer the *robustness principle* and simply ignore, but in this case since the request is potentially harmful giving an error response is not a bad thing.

Reply:
Yeah, I meant that those properties could have been made not accessible in the controller on the fly by calling `setAccess()` on the entity returned by `userEntity()`. Therefore those properties would still be accessible in other contexts, but not here.

Reply:
Ok, we could use entity `setAccess` now and handle exception raise in `SaveEntityAction` in a future refactor?

Reply:
Yes. But if raising an error in this case is a requirement, we might just leave it as it is and take note of a possible future refactor.

Reply:
It's not a strong requirement actually.

We should probably define different validation behaviors on properties access: some accesses could be ignored, some others could be considered dangerous.

Reply:
Problem using  `$entity->setAccess(['username', 'password', 'password_hash', 'email'], false);` is that we than have `username` and `email` in `meta` instead of `attributes`... :thinking: ...

Reply:
solved with `$entity = $this->userEntity()` after `$action`